### PR TITLE
MBS-7611 Migrate documentation from Hugo to MkDocs

### PR DESCRIPTION
**Pull request by eugene-krivobokov**
Migrated documentation from Hugo to MkDocs. 
The theme is https://squidfunk.github.io/mkdocs-material/

Screenshot from 2021-01-19 00-18-15

## The context
In short:
- To make it consistent with DaaS
- Almost all features work just out of the box, smoother and easier to configure: full-text search, mermaid, ...
- Syntax for extensions in markdown looks cleaner and more concise

See a more detailed rationale in the task and linked story. 

## What to check
Sorry for the many changes. Most of them are just deleted files from Hugo's theme.
I've commented on key files. You can check only them.

## In the next PRs

I'll add a linter for markdown. Need to revise implementation. I've decided to postpone it.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -6,8 +6,6 @@ set -e
 
 source $(dirname $0)/_main.sh
 
-docs/check.sh
```
Comment (eugene-krivobokov) [inline build.sh:9]: > I'll add a linter for markdown. Need to revise implementation. I've decided to postpone it.

File: ci/documentation/_publish_internal.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env sh
```
Comment (eugene-krivobokov) [inline ci/documentation/_publish_internal.sh:1]: Script for publishing is almost the same as before.
Git didn't recognize moving with changes.

File: docs/_publish_internal.sh [hunk 3]
```diff
@@ -1,58 +0,0 @@
-#!/usr/bin/env sh
```
Comment (eugene-krivobokov) [inline docs/_publish_internal.sh:1]: Moved file https://github.com/avito-tech/avito-android/pull/715/files#r559796073

File: docs/archetypes/default.md [hunk 4]
```diff
@@ -1,6 +0,0 @@
----
-title: "{{ replace .Name "-" " " | title }}"
-date: {{ .Date }}
-draft: true
----
-
```
Comment (eugene-krivobokov) [inline docs/archetypes/default.md:6]: Most of the deleted files were from Hugo theme

File: docs/mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    social:
+        -   icon: fontawesome/brands/twitter
+            link: https://twitter.com/avitotech
+        -   icon: fontawesome/brands/telegram
+            link: https://t.me/avito_android_opensource
+
+google_analytics:
+    - UA-157613383-1
+    - auto
+
```
Comment (eugene-krivobokov) [inline docs/mkdocs.yml:67]: The main config is here.
The structure of the pages remains the same.

File: docs/mkdocs.yml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    -   Blog:
+            -   2020:
+                    -   03:
+                            -   Testing a minimized build: blog/2020/03/testing-a-minimized-build.md
+                    -   02:
+                            -   Lint Worker API workaround: blog/2020/02/lint-worker-api-workaround.md
+            -   2019:
+                    -   12:
+                            -   Open Source introduction: blog/2019/12/open-source-introduction.md
+                            -   Open source для CI/CD и тестовой инфраструктуры Авито для Android: blog/2019/12/open-source-introduction.ru.md
```
Comment (eugene-krivobokov) [inline docs/mkdocs.yml:104]: Multilingual is not supported yet: https://github.com/mkdocs/mkdocs/issues/774

File: docs/content/contributing/documentation.md [hunk 7]
```diff
@@ -1,8 +1,3 @@
----
-title: Documentation
-type: docs
----
-
```
Comment (eugene-krivobokov) [inline docs/content/contributing/documentation.md:5]: Here you can see most of all the difference in formatting between Hugo and MkDocs

File: ci/documentation/publish.sh [hunk 8]
```diff
@@ -28,16 +28,17 @@ if [[ -z "${GITHUB_GIT_USER_EMAIL+x}" ]]; then
...
         --volume "${SSH_AUTH_SOCK:-/dev/null}":/tmp/ssh_auth_sock \
         --env "SSH_AUTH_SOCK=/tmp/ssh_auth_sock" \
         --env "LOCAL_USER_ID=$(id -u)" \
         --env "GITHUB_GIT_USER_NAME=${GITHUB_GIT_USER_NAME}" \
         --env "GITHUB_GIT_USER_EMAIL=${GITHUB_GIT_USER_EMAIL}" \
         -w="/app" \
-        ${DOCUMENTATION_IMAGE} \
-        sh -c "docs/_publish_internal.sh"
+        "${DOCUMENTATION_IMAGE}" \
+        bash -c "ci/documentation/_publish_internal.sh"
```
Comment (eugene-krivobokov) [inline ci/documentation/publish.sh:44]: Moved scripts for documentation to `ci/documentation` directory to separate the concerns.

**Review by dsvoronin — APPROVED** [review]

**Pull request merged**